script.js

```diff
@@ -266,9 +340,18 @@ function addUpStudentCredits(reportCardTableElement) {
  */
 
 function calculateSemesterGpa(reportCardTableElement) {
-  // code goes here
+  let grades = Array.from(reportCardTableElement.querySelectorAll('.gpa')).filter((grade) => grade.innerHTML in gpaPointsLookup);
```

Comment:
`gpaPointsLookup`  is outer scoped as well. This will work fine for your scenario, but you can save yourself some future debugging pain by explicitly passing all the variables you need for your functions.